src/mbed_tools/cli/project_management.py

```diff
@@ -93,7 +93,7 @@ def _print_dependency_table(libs: List) -> None:
                 lib.reference_file.stem,
                 lib.get_git_reference().repo_url,
                 lib.source_code_path,
-                lib.get_git_reference().ref,
+                "master" if lib.get_git_reference().ref == "" else lib.get_git_reference().ref,
```

Comment:
I don't believe we need to nor should do this. We should clone without specifying a reference when the .lib file has no reference, in order to use the actual default branch for the repo to clone: this may or may not be master. Trying to clone master will fail on repositories that do not use `master` for any branch, and using `master` will not clone the default branch when the default is something other than `master` (as determined by the remote repository).